sdk/mosaic_internal.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2018 ProximaX Limited. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package sdk
+
+import (
+	"errors"
+	"math/big"
+)
+
+func bigIntToMosaicId(bigInt *big.Int) *MosaicId {
+	if bigInt == nil {
+		return nil
+	}
+
+	mscId := MosaicId(*bigInt)
+
+	return &mscId
+}
+
+func mosaicIdToBigInt(mscId *MosaicId) *big.Int {
+	if mscId == nil {
+		return nil
+	}
+
+	return (*big.Int)(mscId)
+}
+
+func generateMosaicId(namespaceName string, mosaicName string) (*MosaicId, error) {
+	if mosaicName == "" {
+		return nil, ErrInvalidMosaicName
+	}
+
+	namespacePath, err := GenerateNamespacePath(namespaceName)
+	if err != nil {
+		return nil, err
+	}
+
+	if !regValidMosaicName.MatchString(mosaicName) {
+		return nil, ErrInvalidMosaicName
+	}
+
+	if bigInt, err := generateId(mosaicName, namespacePath[len(namespacePath)-1]); err != nil {
+		return nil, err
+	} else {
+		return bigIntToMosaicId(bigInt), nil
+	}
```

Comment:
func don't have final return.
 May be, recode lines 44-48 as:

bigInt, err := generateId(mosaicName, namespacePath[len(namespacePath)-1])
if  err != nil {
 		return nil, err
}
return bigIntToMosaicId(bigInt), nil